vllm/model_executor/layers/sampler.py

```diff
@@ -88,12 +102,69 @@ def forward(
         # Sample the next tokens.
         sample_results = _sample(probs, logprobs, input_metadata)
         # Get the logprobs query results.
-        prompt_logprobs, sample_logprobs = _get_logprobs(
-            logprobs, input_metadata, sample_results)
+        prompt_logprobs, sample_logprobs, prompt_dis = _get_logprobs(
+            probs, logprobs, input_metadata, sample_results)
         return _build_sampler_output(sample_results, input_metadata,
-                                     prompt_logprobs, sample_logprobs)
+                                     prompt_logprobs, sample_logprobs, prompt_dis)
 
+    def spec_forward(self,
+        embedding: torch.Tensor,
+        hidden_states: torch.Tensor,
+        input_metadata: InputMetadata,
+        embedding_bias: Optional[torch.Tensor] = None,
+    ) -> SamplerOutput:
```

Comment:
Perhaps you may want to add a comment to tell what is the main difference between original forward and spec_forward